src/index.ts

```diff
@@ -99,26 +99,44 @@ class AuthProvider {
 
     this.iframeWrapper = new IframeWrapper({
       appId: this.appId,
-      iframeUrl: this.state.iframeUrl,
+      iframeUrl: this.networkConfig.walletUrl,
       appConfig: this.appConfig,
       position: position,
     })
 
-    const walletType = await getWalletType(this.appId)
+    const walletType = await getWalletType(
+      this.appId,
+      this.networkConfig.gatewayUrl
+    )
     this.iframeWrapper.setWalletType(walletType, appMode)
 
-    this._provider = new ArcanaProvider(this.iframeWrapper)
-    await this._provider.init()
+    this._provider = new ArcanaProvider(this.iframeWrapper, this.rpcConfig)
+    await this._provider.init({
+      loginWithLink: this.loginWithLink,
+      loginWithSocial: this.loginWithSocial,
+    })
     this.setProviders()
   }
 
   /**
    * A function to trigger social login in the wallet
    */
-  public async loginWithSocial(loginType: string) {
+  loginWithSocial = async (loginType: string) => {
```

Comment:
Just curious to know why this was changed to arrow function?

Reply:
It's being passed in the provider init function, which goes into penpal communication which sort of gets used as message event handler. I could have used `.bind(this)` before passing to make sure it keeps the `this` context, used arrow function as a preemptive measure to not do `.bind(this)` before passing.